nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -134,12 +134,8 @@ impl DataStore {
         //     .authorize(authz::Action::ListChildren, &authz::IP_POOL_LIST)
         //     .await?;
 
-        // join ip_pool to ip_pool_resource and filter
-
-        // used in both success and error outcomes
-        let lookup_type = LookupType::ByCompositeId(
-            "Default pool for current silo".to_string(),
-        );
+        let lookup_type =
+            LookupType::ByOther("default IP pool for current silo".to_string());
```

Comment:
Other than stripping the ID part of the string (which is helpful because it'll not longer look like an error) I don't know that this provides that much more actionable context. Something like this might be more helpful in the logs:

```rust
LookupType::ByQuery(format!("ip_pool where ip_pool_resource { resource_id: {}, is_default: true }", authz_silo_id));
```

It doesn't have to be exactly that. I know this doesn't communicate in plain english the error (as you've done above), but it does communicate some specifics of what tables to look at if something unexpected is happening. 

cc @internet-diglett do you have thoughts?

Reply:
I considered options more like that, giving the stuff we select by, but to a certain extent I think it’s revealing too much about how the concept of default pool is implemented. It feels weird for an API to be talking about a join table by name.

Reply:
How about combining the two? As with other errors in the system, there's both an internal and external message here. Internally we want enough context to know what to dig into if something's gone wrong. Externally we want a general message for the users to know what's failed at a high level.

Reply:
True, interesting idea. Mostly only the 500-level errors have internal messages, except `InvalidValue` is a 400-level one used for things like disk size being too big and it does have that. It would be pretty invasive to change `ObjectNotFound` to have that.

https://github.com/oxidecomputer/omicron/blob/c392c76f0cc8a0bdaca5cf7690c6d95dd071e7db/common/src/api/external/error.rs#L28-L73

Reply:
Changing `ObjectNotFound` might be the right way to do it, but that's not what I specifically had in mind. 

<img width="1017" alt="image" src="https://github.com/oxidecomputer/omicron/assets/3087225/4fa2cbb3-3a82-4f37-b7e5-8f73d6a5ad6d">

In the logs above, there's the external event message as you'd expect but also a serialized version of the actual error object. Even if the "query" part of this never showed up in an error, we'd still see it in the logs (we, at least under the context at which this was initially seen). 

Now, it could be that we really only see errors serialized out like that in the logs under very special circumstances (maybe only saga events?). Were that the case, it may not be worth it.